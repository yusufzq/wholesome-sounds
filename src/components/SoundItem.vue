<template>
	<li class='flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50'>
		<div>
			<RouterLink :to='{name: "sound", params: {ID: sound.documentID}}' class='font-bold block text-gray-600'>{{ sound.modifiedName }}</RouterLink>
			<span class='text-gray-500 text-sm'>{{ sound.displayName }}</span>
		</div>
		<div class='text-gray-600 text-lg'>
			<span class='comments'>
				<i class='fa fa-comments text-gray-600'></i>
				{{ sound.commentCount }}
			</span>
		</div>
	</li>
</template>

<script>
	export default {
		name: 'SoundItem',
		props: ['sound']
	};
</script>