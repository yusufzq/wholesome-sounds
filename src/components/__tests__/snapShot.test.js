import { RouterLinkStub, shallowMount } from '@vue/test-utils';
import SoundItem from '@/components/SoundItem.vue';

describe('snapShot', () => {
	it('snapShot SoundItem', () => {
		const sound = {
			documentID: 'testID',
			displayName: 'testDisplayName',
			modifiedName: 'testModifiedName',
			commentCount: 7
		};

		const wrapper = shallowMount(SoundItem, {
			propsData: { sound },
			global: {
				stubs: {RouterLink: RouterLinkStub}
			}
		});

		expect(wrapper.element).toMatchSnapshot();
	});
});