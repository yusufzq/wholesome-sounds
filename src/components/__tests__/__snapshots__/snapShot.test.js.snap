// ViTest SnapShot v1: https://vitest.dev/guide/snapshot.html

exports[`router > parse RouterLink 'to' attribute 1`] = `
	<li
		class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
		id="testID"
	>
		<div>
			<a
				class="font-bold block text-gray-600"
			>
				testModifiedName
			</a>
			<span
				class="text-gray-500 text-sm"
			>
				testDisplayName
			</span>
		</div>
		<div
			class="text-gray-600 text-lg"
		>
			<span
				class="comments"
			>
			<i
				class="fa fa-comments text-gray-600"
			/>
				7
			</span>
		</div>
	</li>
`;